docs/src/pages/components/data-grid/rows/InfiniteLoadingGrid.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import { XGrid, GridOverlay } from '@material-ui/x-grid';
+import { useDemoData } from '@material-ui/x-grid-data-generator';
+import LinearProgress from '@material-ui/core/LinearProgress';
+
+function CustomLoadingOverlay() {
+  return (
+    <GridOverlay>
+      <div style={{ position: 'absolute', top: 0, width: '100%' }}>
+        <LinearProgress />
+      </div>
+    </GridOverlay>
+  );
+}
+
+export default function InfiniteLoadingGrid() {
+  const { loading, data, setRowLength, loadNewData } = useDemoData({
+    dataSet: 'Commodity',
+    rowLength: 20,
+    maxColumns: 6,
+  });
+
+  const handleOnRowsScrollEnd = (params) => {
+    const newRowLength = params.virtualRowsCount + params.viewportPageSize;
+    setRowLength(newRowLength);
+    loadNewData();
```

Comment:
Why are we calling the two methods? Looking at the source of `useDemoData.ts`, they seem to exclude themselves. 

```suggestion
    setRowLength(newRowLength);
```

Reply:
On a different note. The new data is loaded so quickly it doesn't feel real. What do you think about https://github.com/mui-org/material-ui-x/blob/f4c167b85841b38fddeec021ea7fe7e1d6991c98/docs/src/pages/components/data-grid/filtering/ServerFilterGrid.tsx#L29?

Reply:
added